python packaging fails without Internet access

Trying to compile the latest syslog-ng git snapshot I ran into a problem: the build process tries to use the Internet. There is no Internet access in the openSUSE build service, and turned off by default in the Fedora COPR build service. FreeBSD packages are also built without Internet access. I do not follow Debian / Ubuntu closely, but AFAIR they do not allow Internet access either.

```
[  107s] if [ "venv" = "venv" ]; then \
[  107s]     ./scripts/build-python-venv.sh "/usr/bin/python3" "/home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.248.gbc48493/venv" "."; \
[  107s] fi
[  107s] Building dev virtualenv for syslog-ng at /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.248.gbc48493/venv from ./dev-requirements.txt
[  112s] WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x56128a67a728>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/cachetools/
[  113s] WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x56128a6dd648>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/cachetools/
[  114s] WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x56128a6cc748>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/cachetools/
[  116s] WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x56128a5d4f48>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/cachetools/
[  120s] WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x56128a6e2728>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/cachetools/
[  120s] ERROR: Could not find a version that satisfies the requirement cachetools==4.2.4 (from -r ./requirements.txt (line 1)) (from versions: none)
[  120s] ERROR: No matching distribution found for cachetools==4.2.4 (from -r ./requirements.txt (line 1))
[  120s] make[4]: *** [Makefile:29295: /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.248.gbc48493/.python-venv-built] Error 1
[  120s] make[3]: *** [Makefile:28203: install-exec-am] Error 2
[  120s] make[2]: *** [Makefile:26911: install-am] Error 2
[  120s] make[1]: *** [Makefile:25118: install-recursive] Error 1
[  120s] make: *** [Makefile:26904: install] Error 2
[  120s] error: Bad exit status from /var/tmp/rpm-tmp.dQhhv2 (%install)
```

Now, that Kubernetes support is merged, package building fails again. It would be pretty useful, if turning off building/packaging Python packages would be possible (just like with Java). For Linux packagers who do not want to enable them, or BSD packagers, where there is no Kubernetes support at all, but Python is in active use. Unfortunately, build fails with:

```
[   90s] (cd ./modules/python-modules && \
[   90s]  mkdir -p /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules && \
[   90s]          /usr/bin/python3 setup.py egg_info --egg-base="/home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules"  && \
[   90s]          if [ -f /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/requires.txt ]; then /usr/bin/python3 -m pip install -r /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/requires.txt; fi)
[   91s] running egg_info
[   91s] creating /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info
[   91s] writing /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/PKG-INFO
[   91s] writing dependency_links to /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/dependency_links.txt
[   91s] writing requirements to /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/requires.txt
[   91s] writing top-level names to /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/top_level.txt
[   91s] writing manifest file '/home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/SOURCES.txt'
[   91s] reading manifest file '/home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/SOURCES.txt'
[   91s] writing manifest file '/home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/SOURCES.txt'
[   91s] Defaulting to user installation because normal site-packages is not writeable
[   91s] WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x557c4fff0f08>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/kubernetes/
[   92s] WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x557c4ffb3cb8>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/kubernetes/
[   93s] WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x557c500079b8>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/kubernetes/
[   95s] WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x557c4ff3aff8>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/kubernetes/
[   99s] WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x557c4ffd9408>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /simple/kubernetes/
[   99s] ERROR: Could not find a version that satisfies the requirement kubernetes (from -r /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/requires.txt (line 1)) (from versions: none)
[   99s] ERROR: No matching distribution found for kubernetes (from -r /home/abuild/rpmbuild/BUILD/syslog-ng-3.38.1.344.gd829200/modules/python-modules/syslogng.egg-info/requires.txt (line 1))
[   99s] make[4]: *** [Makefile:29633: pymodules-install-requirements] Error 1
[   99s] make[3]: *** [Makefile:28238: install-exec-am] Error 2
[   99s] make[2]: *** [Makefile:26945: install-am] Error 2
[   99s] make[1]: *** [Makefile:25152: install-recursive] Error 1
[   99s] make: *** [Makefile:26938: install] Error 2
```

--- a/configure.ac
+++ b/configure.ac
@@ -339,6 +339,9 @@ AC_ARG_ENABLE(python,
               [  --disable-python       Disable Python module]
               ,,enable_python="auto")
 
+AC_ARG_ENABLE(python-modules,
+              [  --disable-python-modules  Disable Python based syslog-ng modules]
+              ,,enable_python_modules="auto")
 
 AC_ARG_WITH(python,
               [  --with-python=VERSION    Build with a specific version of python]
@@ -1532,6 +1535,13 @@ if test "x$enable_python" = "xyes"; then
     fi
 fi
 
+dnl ***************************************************************************
+dnl python modules
+dnl ***************************************************************************
+
+if test "x$enable_python_modules" = "xauto"; then
+    enable_python_modules=$enable_python
+fi
 
 dnl ***************************************************************************
 dnl docbook
@@ -1961,6 +1971,7 @@ AM_CONDITIONAL(LIBRABBITMQ_INTERNAL, [test "x$with_librabbitmq_client" = "xinter
 AM_CONDITIONAL(ENABLE_RIEMANN, [test "$enable_riemann" != "no"])
 AM_CONDITIONAL(ENABLE_JOURNALD, [test "$with_systemd_journal" != "no"])
 AM_CONDITIONAL(ENABLE_PYTHON, [test "$enable_python" != "no"])
+AM_CONDITIONAL(ENABLE_PYTHON_MODULES, [test "$enable_python_modules" != "no"])
 AM_CONDITIONAL(ENABLE_JAVA, [test "$enable_java" = "yes"])
 AM_CONDITIONAL(ENABLE_JAVA_MODULES, [test "$enable_java_modules" = "yes"])
 AM_CONDITIONAL(ENABLE_AFSNMP, [test "$enable_afsnmp" = "yes"])
@@ -2104,8 +2115,9 @@ echo "  GEOIP2 support (module)     : ${enable_geoip2:=no}"
 echo "  Redis support (module)      : ${enable_redis:=no}"
 echo "  Kafka support (module)      : ${enable_kafka:=no}"
 echo "  Riemann destination (module): ${enable_riemann:=no}, microseconds: ${riemann_micros:=no}"
-echo "  python                      : ${enable_python:=no} (pkg-config package: ${with_python:=none}, packages: ${with_python_packages})"
-echo "  java                        : ${enable_java:=no}"
-echo "  java modules                : ${enable_java_modules:=no}"
+echo "  Python                      : ${enable_python:=no} (pkg-config package: ${with_python:=none}, packages: ${with_python_packages})"
+echo "  Python modules              : ${enable_python_modules:=no}"
+echo "  Java                        : ${enable_java:=no}"
+echo "  Java modules                : ${enable_java_modules:=no}"
 echo "  afsnmp module               : ${enable_afsnmp:=no}"
 echo "  native bindings             : ${enable_native:=no}"

--- a/modules/python-modules/CMakeLists.txt
+++ b/modules/python-modules/CMakeLists.txt
@@ -1,3 +1,12 @@
+module_switch(ENABLE_PYTHON_MODULES "Enable Python based modules" ENABLE_PYTHON)
+
+if (NOT ENABLE_PYTHON_MODULES)
+  return()
+endif ()
+
+if (NOT ENABLE_PYTHON)
+  message(FATAL_ERROR "Python based modules enabled, but the required syslog-ng Python plugin is disabled.")
+endif()
 
 add_custom_target(BuildPyModules
     COMMAND ${PYTHON_VENV_EXECUTABLE} setup.py build --build-base=${CMAKE_CURRENT_BINARY_DIR}/build

--- a/modules/python-modules/Makefile.am
+++ b/modules/python-modules/Makefile.am
@@ -52,7 +52,7 @@ EXTRA_DIST += \
 	modules/python-modules/setup.py						\
 	modules/python-modules/README.md
 
-if ENABLE_PYTHON
+if ENABLE_PYTHON_MODULES
 
 python_sysconf_module_DATA = modules/python-modules/README.md
 

